fix: Updated the docs for scorecard feature in IDP (#3688)
harness/developer-hub: docs/internal-developer-portal/features/scorecard.md

<docimage path={require('../../internal-developer-portal/features/static/concept-scorecard.png')}/>

- **Check**: A check is a query performed against a data point for a software component which results in either `Pass` or `Fail`.

> - **Data Source**: Data Sources are third-party providers which can provide a specific type of data for a software component. Data sources are fixed in IDP and have a specific way of enablement. 

Reviewer: ```suggestion
- **Data Source**: Data Sources are third-party providers which can provide a specific type of data for a software component. Example - GitHub, GitLab, Harness, Pagerduty, etc.
```

Reviewer: Not sure about the "fixed" part. We'll be adding new data sources and in future, custom data source as well.

> - **Data Points**: For each software componenet, every data source provides some data points. The data points could be a number, a string or a boolean. 

Reviewer 2: ```suggestion
- **Data Points**: For each software component, every data source provides some data points. The data points could be a number, a string or a boolean. 
```

</details>

> #### Enable Scorecards

Reviewer: ```suggestion
## Enable Scorecards
```

Isn't this a larger section of this page given the section below with 3 # seems like a child of the enable scorecards section?

> #### Enable Scorecards

Reviewer: Can we have a "disable scorecards" as well?

Plus I feel "How to enable Scorecards in IDP" is probably more descriptive and catches the attention.

Reviewer: Makes sense - thank you Deba!

### Scorecard Components in IDP

Scorecard has two main UI components which are developer facing and lives in the Catalog - 1. A small `Card` for the Overview page with the scores and, 2. A `Tab` view with details of the checks and how the score got computed. This is illustrated below. The Tab view contains detailed comprehensive information as shown in the image under [overview](/docs/internal-developer-portal/features/scorecard#overview)

<docimage path={require('../../internal-developer-portal/features/static/scorecard-overviewpage.png')}/>

:::info

> For the steps **1** and **2** the `Layout` is autopopulated with the adequate fields in the recent version of IDP and could be skipped, but users are still advised to check for the same on their `Layout` for the software components.  

Reviewer: ```suggestion
For the steps **1** and **2** the `Layout` is prepopulated with the adequate fields in the recent version of IDP and could be skipped, but users are still advised to check for the same on their `Layout` for the software components.  
```
